### PR TITLE
protect against missing service offering

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #9160 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Review by harikrishna-patnala — APPROVED** [review]
code LGTM

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]
code lgtm

File: server/src/main/java/com/cloud/api/query/dao/VolumeJoinDaoImpl.java [hunk 1]
```diff
@@ -283,6 +283,19 @@ public VolumeResponse newVolumeResponse(ResponseView view, VolumeJoinVO volume)
         return volResponse;
     }
 
+    private static ServiceOffering getServiceOfferingForDiskOffering(VolumeJoinVO volume, DiskOffering computeOnlyDiskOffering) {
+        ServiceOffering serviceOffering = null;
+        // first try existing ones
+        if (computeOnlyDiskOffering != null) {
+            serviceOffering = ApiDBUtils.findServiceOfferingByComputeOnlyDiskOffering(volume.getDiskOfferingId(), false);
+        }
+        // else try removed ones
```
Comment (BryanMLima) [inline server/src/main/java/com/cloud/api/query/dao/VolumeJoinDaoImpl.java:292]: @DaanHoogland, could you change this comment to the Javadoc of the method?

**Review by BryanMLima — COMMENTED** [review]
Just one comment, overall, LGTM.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by vladimirpetrov — APPROVED** [review]
LGTM based on manual testing, couldn't get a NPE even after nasty db manipulations.

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Pull request merged**